pull request 20241114

lztiancn opened the pull request:
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

lztiancn committed:
feat: add fastapi requirement file

lztiancn committed:
feat: docker增加fastapi

lztiancn committed:
feat: 修改dockerfile

lztiancn committed:
fix: dockerrun调通fastapi

lztiancn committed:
Merge branch 'master' into master

lztiancn committed:
fix: docker compose 启动fastapi

lztiancn committed:
Merge branch 'master' of https://github.com/lztiancn/lzmineru

lztiancn committed:
fix: 增加redis部署

lztiancn committed:
fix: 增加redis部署

lztiancn committed:
feat: 增加异步处理

lztiancn committed:
fix: 优化处理队列

nil-andreu commented on Dockerfile:
```diff
@@ -41,10 +44,17 @@ RUN /bin/bash -c "wget https://gitee.com/myhloli/MinerU/raw/master/magic-pdf.tem
     sed -i 's|cpu|cuda|g' /root/magic-pdf.json"
 
+# install extents
+COPY requirements-fastapi.txt /minerugw/requirements-fastapi.txt
```
To do not increase by default the size of the image, this could be set as a optional requirements.

lztiancn committed:
fix: 增加md5查询功能

lztiancn committed:
fix: 增加url取不到数据的异常处理

Commit:
增加fastapi

Commit:
增加接口文档说明

Commit:
增加贡献者lztiancn说明

Commit:
.gitignore

Commit:
保存pdf文件并在重启后重新加载队列

Commit:
重建任务队列时恢复初始状态
返回当前队列长度

Commit:
app里面需要临时写images

Commit:
Merge remote-tracking branch 'upstream/master'

myhloli committed:
Update pdf_parse_union_core_v2.py

myhloli committed:
build(deps): add upper version limit for PyMuPDF

- Set PyMuPDF version to <= 1.24.14 in all requirements files
- Prevent potential compatibility issues with future versions

myhloli committed:
feat(layout): improve title block handling and layout detection

- Merge title blocks that are close to each other horizontally
- Adjust line insertion logic for title blocks- Increase image size and decrease confidence threshold for layout detection
- Update DocLayoutYOLO model weights
- Refactor drawing of bounding boxes for different block types

myhloli committed:
refactor(BatchAnalyze): comment out image rotation logic in doclayout_yolo

myhloli committed:
feat(post_proc): enhance title block processing with average line height

- Add average line height calculation for title blocks
- Include page number in title dictionary
- Improve title optimization prompt for better hierarchy- Implement retry mechanism for JSON decoding errors
- Add error logging for title count mismatch

myhloli committed:
refactor(pre_proc): adjust IOU threshold for character overlap detection

- Modified the IOU threshold in ocr_span_list_modify.py from 0.9 to 0.35
- This change aims to improve the detection of overlapping characters in OCR processed PDFs

myhloli committed:
docs(magic_pdf): update llm_aided.py prompt for title list optimization

- Clarify the expected format for the optimized title list JSON output- Emphasize the need to return only the title levels in the specified format

myhloli committed:
fix(language): remove invalid UTF-16 surrogate pairs from input text

- Add `remove_invalid_surrogates` function to filter out invalid UTF-16 surrogate pairs
- Integrate the new function into the `detect_lang` workflow
- Include a test case with UTF-16 surrogates to verify the fix

myhloli committed:
update logo

myhloli committed:
build(docker): update doclayout-yolo dependency

- Remove doclayout_yolo==0.0.2b1 and doclayout-yolo==0.0.2
- Add doclayout-yolo==0.0.2b1 to all requirements files

myhloli committed:
feat(model): improve batch analysis logic and support npu

- Add support for NPU (Neural Processing Unit) when available
- Implement batch analysis for GPU and NPU devices
- Optimize memory usage and improve performance
- Update logging and error handling

myhloli committed:
refactor(magic_pdf): improve title block merging logic

- Rename and update merge_title_blocks function
- Implement merge_two_bbox helper function
- Refactor merging logic to preserve original block structure- Update function calls and integrate with existing pipeline

Commit:
add fork

Commit:
Merge remote-tracking branch 'upstream/master'

myhloli committed:
fix(magic_pdf): correct end page index and improve error handling

- Adjust end_page_id calculation to prevent IndexError when accessing pages
- Enhance error handling in LLM post-processing by specifically catching JSONDecodeError

myhloli committed:
docs(README): update demo badges

- Update OpenDataLab badge to new design

myhloli committed:
docs(README): update demo badges

- Update OpenDataLab badge to new design

myhloli committed:
feat(table): upgrade RapidTable to1.0.3 and add sub-model support

- Update RapidTable dependency to version 1.0.3
- Add support for sub-models in RapidTable
- Update magic-pdf configuration to include table sub-model
- Modify table model initialization to support sub-models
- Update table prediction logic to handle new output format

myhloli committed:
build(docker): update rapid-table dependency

- Update rapid-table version from ==0.3.0 to >=1.0.3,<2.0.0 in multiple requirements files
- This change affects Ascend NPU, China, and Global Docker configurations

myhloli committed:
refactor(model): update batch analyze logic for rapid table model

- Modify the batch analyze process to handle the rapid table model's output
- Add logic_points variable to capture additional output from rapid table prediction

myhloli committed:
docs(README): update WeChat group link

- Update WeChat group link in both README.md and README_zh-CN.md

myhloli committed:
refactor(table): add device configuration for Unitable model

- Import get_device function from magic_pdf.libs.config_reader- Update RapidTableModel initialization to include device parameter for Unitable model

myhloli committed:
refactor(model): update config version check to 1.1.1

- Update the version check in download_models.py and download_models_hf.py
- Change the threshold from '1.1.0' to '1.1.1' for model configuration updates

myhloli committed:
fix(magic_pdf): limit batch ratio for GPU memory

- Commented out the original batch ratio calculation
- Set a fixed batch ratio of 2 for GPUs with less than 8 GB memory
- Increased batch ratio to 4 for GPUs with 8 GB or more memory

myhloli committed:
feat(llm_aided): add reasonability check and fine-tuning guidelines

- Added instructions for checking the reasonability of heading levels
- Included guidelines for making fine adjustments based on context and logic
- Emphasized the importance of aligning the final result with the document's actual structure

moria97 committed:
Fix ocr utills

github-actions[bot] committed:
@moria97 has signed the CLA in opendatalab/MinerU#1578

myhloli committed:
feat(pdf_parse): remove tilted lines for better text extraction

- Add remove_tilted_line function to filter out lines with angles between 2 and 88 degrees
- Integrate the new function into the text extraction process
- Improve the accuracy of text block processing by removing non-horizontal/vertical lines

myhloli committed:
fix(ocr): improve ONNX model initialization and error handling

- Add key length validation for ONNX model initialization
- Move import statements to the top of the file
- Wrap model initialization in a try-except block for better error handling
- Refactor code to improve readability and maintainability

myhloli committed:
fix(models): update unimernet_small model path

- Update model path from 'unimernet_small' to 'unimernet_small_2501' in multiple scripts and configuration files
- This change affects download_models.py, download_models_hf.py, and model_configs.yaml

myhloli committed:
perf(model): adjust batch size for layout and formula detection

- Reduce YOLO_LAYOUT_BASE_BATCH_SIZE from 4 to 1
- Simplify batch ratio calculation for formula detection
- Remove unused conditional logic in batch ratio determination

myhloli committed:
perf(magic_pdf): optimize batch ratio calculation for GPU

- Update GPU memory check and batch ratio calculation logic
- Add support for virtual VRAM size environment variable
- Improve logging for GPU memory and batch ratio

myhloli committed:
refactor(magic_pdf): adjust VRAM allocation and MFR batch size- Update VRAM allocation logic to use 'VIRTUAL_VRAM_SIZE' environment variable
- Reduce MFR (Math Formula Recognition) batch size from 64 to 32

myhloli committed:
perf(magic_pdf): adjust batch ratio calculation for GPU memory

- Reduce batch_ratio by 1 for better performance and stability
- This change ensures more consistent memory usage when processing documents

myhloli committed:
perf(magic_pdf): optimize batch processing for GPU

- Improve batch ratio calculation based on GPU memory
- Enhance performance for devices with 8GB or more VRAM

myhloli committed:
fix(magic_pdf): correct batch ratio conditions for GPU memory

- Update conditions for batch ratio assignment:
  -8 <= gpu_memory < 10: batch_ratio = 2 - 10 <= gpu_memory <= 12: batch_ratio =4
- This fix ensures proper batch ratio selection for GPU memory sizes

myhloli committed:
refactor(pdf_parse): uncomment char bbox validation logic

- Restore commented code for filtering out characters with invalid bounding boxes
- This change may affect the filtering of unnecessary characters in PDF parsing

myhloli committed:
fix(boxbase): handle cases where bounding box area is zero

- Add a check to return 0 when either bbox1_area or bbox2_area is zero
- This prevents division by zero errors when calculating IoU

myhloli committed:
feat(pdf_parse_union_core_v2): add timing log for LLM aided processes

- Add timing measurement for formula, text, and title optimization using LLM
- Log the execution time for each LLM aided process

myhloli committed:
docs(readme):update readme for 1.1.0

myhloli committed:
docs(url): update Miners links in header

- Change Miners homepage link from 'https://mineru.org.cn/home?source=online' to 'https://mineru.net/home?source=online'
- Change Miners client download link from 'https://mineru.org.cn/client?source=online' to 'https://mineru.net/client?source=online'

myhloli committed:
feat(table-config): add sub_model configuration for rapid_table

- Add sub_model configuration option for rapid_table model
- Provide two sub_model options: slanet_plus and unitable

myhloli committed:
docs(readme): update changelog for v1.1.0 release- Update model capabilities: upgrade to latest doclayout_yolo(2501) and unimernet(2501) models

- Improve performance: optimize resource usage and processing pipeline for faster parsing on high-end devices- Enhance parsing effects: add new heading classification feature to online demo
- Refactor changelog structure for better readability and organization

myhloli committed:
docs(README): update online demo links and enhance documentation readability

- Update online demo links in both English and Chinese README files

myhloli committed:
Update version.py with new version

myhloli committed:
Update python-package.yml

myhloli committed:
Update README.md

myhloli committed:
Update README_zh-CN.md

futuremeng committed:
merge: lztiancn/lzmineru

futuremeng committed:
fix: context: ./docker/china

lztiancn committed:
feat: 增加回调

lztiancn committed:
Merge remote-tracking branch 'upstream/master'

lztiancn committed:
fix: 修改解析错误DiskReaderWriter

lztiancn committed:
fix: 修改解析错误image_dir

lztiancn committed:
feat: 发送回调使用异步

lztiancn committed:
fix: 代码优化

HunterLG reviewed (approved):
Adapts to Redis to implement status persistence and project implementation.
